### PR TITLE
Added support for SSEKMSKeyId in encrypt-keys action

### PR DESCRIPTION
**Pull request by calvinhp**
This adds the ability to encrypt s3 buckets using a specific KMS key. This addresses issue #562

### CODE REVIEW

**Comment by kapilt** [comment]
the travis build failure is signifying the jsonschema wasn't updated to reflect the new options. export C7N_VALIDATE=yes when running tests to trigger locally, probably need that in the dev docs, i had it disabled because it increased the test execution runtime, but given this is the second pull request in as many days thats had the issue it might be worth turning on by default.

**Comment by kapilt** [comment]
re slow down in tests, its about 30% slower at the moment with the option enabled, but realistically its only 3.2s additional

```
AWS_ACCESS_KEY_ID=foo AWS_SECRET_ACCESS_KEY=bar AWS_DEFAULT_REGION=us-east-1 ./bin/nosetests  --processes=-1 tests
..................................................................................................................................................................................................................................................................................................................................................................................................................................................
----------------------------------------------------------------------
Ran 434 tests in 8.948s

OK
(custodian):custodian kapilt$ export C7N_VALIDATE=yes
(custodian):custodian kapilt$ make test                                                                                            
AWS_ACCESS_KEY_ID=foo AWS_SECRET_ACCESS_KEY=bar AWS_DEFAULT_REGION=us-east-1 ./bin/nosetests  --processes=-1 tests
..................................................................................................................................................................................................................................................................................................................................................................................................................................................
----------------------------------------------------------------------
Ran 434 tests in 12.257s

OK
```

**Comment by kapilt** [comment]
the other downside on the policy to test policies, it makes testing much slower, with the merge of this full test run goes over 10s, with the test here landing at #2 on the slow test list. tests.test_s3.S3Test.test_encrypt_keys: 0.3278s

`make ttest` for test timings. its definitely something i track, we want to keep tests fast and light whenever possible so we do more of it ;-) good fight in a lost cause sort of thing.

File: tests/data/placebo/test_s3_encrypt/s3.CopyObject_1.json [hunk 1]
```diff
@@ -3,21 +3,31 @@
...
             "ETag": "\"c68271a63ddbc431c307beb7d2918275\""
         }, 
         "ResponseMetadata": {
             "HTTPStatusCode": 200, 
-            "HostId": "L1ZMtrOGwF+vkecrlREyjgBXICb17QQzgslMtlD2cuw/wPfW2WQdduJCdv0SihxDegQqekKx56g=", 
-            "RequestId": "DBCAF1251C0F0B79"
+            "RetryAttempts": 0, 
+            "HostId": "YrgyrXKdjgYLs09ixH8UUOpZMhK7VaqpoV/7lZZjIUr3VY2VD4SP0NW2h+/nNW9lI8sK5dYzUX8=", 
+            "RequestId": "D4AABBE372B7F5EB", 
+            "HTTPHeaders": {
```
Comment (calvinhp) [inline tests/data/placebo/test_s3_encrypt/s3.CopyObject_1.json]: @kapilt Should I have modified the output from placebo to remove the `HTTPHeaders` to make these diffs smaller. (for future reference)

Comment (kapilt) [inline tests/data/placebo/test_s3_encrypt/s3.CopyObject_1.json]: nah i wouldn't worry about the headers, occasionally its nice to strip out extraneous buckets in the reply, but the biggest gain is just avoiding unesc api calls when possible vs stripping individual call outputs. again its fine for this pr, just want awareness for future.

**Pull request merged**